n2y/blocks.py

```diff
@@ -604,6 +606,389 @@ def to_pandoc(self):
+            parent = children.copy_to(parent, False)
+            start = len(parent.children) - len(children)
+            self.children_appended[index]['children'] = parent.children.block_list[start:]
+
+    def _retrieve_node(self, node_id):
+        node = self.client.get_page_or_database(node_id)
+        if node is not None:
+            return node
+        return self.client.get_block(node_id, self.page)
+
+    def _probe_destination_links(self):
+        if '_block' in self.destination.__dict__:
+            for child in self.destination.block.children:
+                self._find_links(child)
+        elif 'schema' in self.destination.__dict__:
+            self._database_error()
+        else:
+            for child in self.destination.children:
+                self._find_links(child)
+
+    def _database_error(self):
+        raise NotImplementedError(
+            'This functionality is not yet supported for Databases'
+        )
+
+    def _find_links(self, child):
+        if isinstance(child, LinkToPageBlock):
+            self._store_link_info(child.notion_data)
+        elif isinstance(child, ParagraphBlock):
+            for text in child.notion_data['rich_text']:
+                if text['type'] == 'mention' and \
+                text['mention']['type'] in ['page', 'database']:  # noqa: E122
+                    self._store_link_info(text['mention'])
+        if child.children:
+            for child in child.children:
+                self._find_links(child)
+
+    def _store_link_info(self, notion_data):
+        title, link_id = self._retrieve_link_info(notion_data)
+        if title != self.default_link_title and title not in self.link_dict:
+            self.link_dict[title] = link_id
+
+    def _retrieve_link_info(self, notion_data):
+        node_id = notion_data[notion_data['type']] if\
+            isinstance(notion_data[notion_data['type']], str) else\
+            notion_data[notion_data['type']]['id']
+        if "page" in notion_data['type']:
+            node = self.client.get_page(node_id)
+        elif "database" in notion_data['type']:
+            node = self.client.get_database(node_id)
+        else:
+            raise NotImplementedError(f"Unknown link type: {notion_data['type']}")
+        if node is None:
+            msg = "Permission denied when attempting to access linked node [%s]"
+            logger.warning(msg, node_id)
+            return self.default_link_title, node_id
+        else:
+            return node.title.to_plain_text(), node_id
+
+    def _define_page(self):
+        if 'schema' in self.destination.__dict__:
+            self._database_error()
+        elif '_block' in self.destination.__dict__:
+            self.page = self.destination
+        else:
+            self.page = self.destination.page
+        if not self.page:
+            raise NotImplementedError(
+                'Page must be defined.'
+            )
+
+    def _default_info(self):
```

Comment:
All of this should really go in `__init__` so that people reading the class can see where all of the properties get defined.